lte/gateway/c/session_manager/SessionStateEnforcer.cpp

```diff
@@ -0,0 +1,238 @@
+      session_state->set_stored_session_version(
+		      cfg.rat_specific_context.m5gsm_session_context().fsm_version());
+   }
+   return true; // for compilation only.
+}
+
+bool SessionStateEnforcer::handle_session_init_rule_updates(
+		SessionMap& session_map, const std::string& imsi,
+   		SessionState&  session_state)
+{
+
+   std::cerr <<__LINE__ << " " << __FUNCTION__ <<" imsi "<< imsi <<"\n";
+   auto itp = pdr_map_.equal_range(imsi);
+   for ( auto itr = itp.first;  itr!= itp.second; itr++) {
+	/* Get the PDR numbers, now  get the rules from global static rule 
+         * list
+         */
+        SetGroupPDR rule;
+		GlobalRuleList.get_rule(itr->second,&rule);
+        std::cerr <<__LINE__ << " " << __FUNCTION__ <<" for imsi " << imsi
+			 << "matched  pdr no "   <<rule.pdr_id()<< "\n";
+		session_state.insert_with_static_rules (&rule);
+	}
+   auto itf = far_map_.equal_range(imsi);
+   for ( auto itr = itf.first;  itr!= itf.second; itr++) {
+	/* Get the PDR numbers, now  get the rules from global static rule 
+         * list
+         */
+        SetGroupFAR rule;
+	GlobalRuleList.get_rule(itr->second,&rule);
+        std::cerr <<__LINE__ << " " << __FUNCTION__ <<" for imsi "<< imsi
+					<<"matched far no " <<rule.far_id()<<"\n";
+        // Add to the the session vector
+	session_state.insert_with_static_rules (&rule);
+    }
+#if 1
+    //auto ip_addr = session_state.get_config().common_context.ue_ipv4();
+    auto ip_addr = session_state.get_config().rat_specific_context.
+               m5gsm_session_context().pdu_address().redirect_server_address();
+    SessionState::SessionCommonInfo  sess_comm_info;
+    sess_comm_info.imsi_ = imsi;
+    sess_comm_info.ip_addr_ = ip_addr;
+    sess_comm_info.Pdr_rules_ = session_state.get_5g_static_pdr_rules();
+    sess_comm_info.Far_rules_ = session_state.get_5g_static_far_rules();
+
+    /* session_state elments are filled with rules. State needs to be 
+     * moved to CREATED and sending message to UPF.
+     * Note: charging and credit related info not taken care in drop-1  
+     */
+    auto update_criteria = get_default_update_criteria();
+    session_state.set_fsm_state(CREATED, update_criteria);
+    MLOG(MDEBUG) << "State of session changed to "
+                 << session_fsm_state_to_str(session_state.get_state());
+    std::cerr <<__LINE__ << " " << __FUNCTION__ <<" changed the state to "
+              << session_fsm_state_to_str(session_state.get_state())
+              << " IP_Address ipv4 " << ip_addr << "of IMSI " << imsi << "\n";
+
+    std::cerr <<__LINE__ << " " << __FUNCTION__ <<" Beofre calling set_upf_session" <<"\n";
+    pipelined_client_->set_upf_session(sess_comm_info,call_back_void_upf);
+#endif
+return true;
+}
+
+bool SessionStateEnforcer:: static_rule_init () {
+// Static PDR, FAR, QDR, URR and BAR mapping  and also
+//  Define 1 PDR and FAR
+   SetGroupPDR reqpdr1,reqpdr2;
+   SetGroupFAR reqf;
+   magma::PDI pdireq;
+#if 0

```

Comment:
What are these enum conditionals? If it is to case between set / crud or 5g/4g, I would rather explicitly create a flag for this.

Reply:
It will be cleaned up.

Reply:
Cleaned up